src/lib/core/fiber_channel.c

```diff
@@ -325,10 +357,10 @@ fiber_channel_put_msg_timeout(struct fiber_channel *ch,
 			 */
 
 			/*
-			 * Closed channels, are, well, closed,
+			 * Destroyed channels, are, well, destroyed,
 			 * even if there is space in the buffer.
 			 */
-			if (ch->is_closed) {
+			if (ch->is_destroyed) {
 				diag_set(ChannelIsClosed);
 				return -1;
 			}
```

Comment:
You added an exit if `ch->is_closed` and there is no yield in between the check and this hunk. So the channel can't be destroyed at this point. Let's replace with an assertion.

Reply:
Done.